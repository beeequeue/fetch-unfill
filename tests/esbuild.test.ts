import path from "node:path"

import esbuild from "esbuild"
import { describe, expect, it } from "vitest"

import { createTester } from "./utils"

const test = createTester(
  "esbuild",
  async (name: string, extraOptions: esbuild.BuildOptions | null = {}) => {
    const result = await esbuild.build({
      entryPoints: [path.resolve(__dirname, "fixtures", `${name}.mjs`)],
      write: false,
      nodePaths: [path.resolve(__dirname, "..", "node_modules")],

      format: "cjs",
      platform: "node",
      target: "node20",
      bundle: true,
      minify: true,

      ...extraOptions,
    })

    expect(result.errors).toStrictEqual([])

    return result.outputFiles![0].text
  },
)

const alias = {
  alias: {
    "node-fetch": "fetch-unfiller/node",
    "cross-fetch": "fetch-unfiller/node",
  },
} satisfies esbuild.BuildOptions

describe("node-fetch", () => {
  it("bundles the original package", async () => {
    await test("node-fetch", null)
  })

  it("unfills it", async () => {
    await test("node-fetch", alias)
  })
})

describe("cross-fetch", () => {
  it("bundles the original package", async () => {
    await test("cross-fetch", null)
  })

  it("unfills it", async () => {
    await test("cross-fetch", alias)
  })
})
