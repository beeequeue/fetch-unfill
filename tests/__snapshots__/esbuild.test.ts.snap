// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cross-fetch > bundles the original package 1`] = `"302.6KB"`;

exports[`cross-fetch > unfills it 1`] = `"629B"`;

exports[`node-fetch > bundles the original package 1`] = `"90.35KB"`;

exports[`node-fetch > unfills it 1`] = `"629B"`;
